internal/operator/clusterutilities_test.go

```diff
@@ -168,3 +174,117 @@ func TestOverrideClusterContainerImages(t *testing.T) {
 		delete(ContainerImageOverrides, config.CONTAINER_IMAGE_CRUNCHY_POSTGRES_GIS_HA)
 	})
 }
+
+func TestGetPgbackrestBootstrapS3EnvVars(t *testing.T) {
+
+	// create a fake client that will be used to "fake" the initialization of the operator for
+	// this test
+	fakePGOClient, err := fakekubeapi.NewFakePGOClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+	// now initialize the operator using the fake client.  This loads various configs, templates,
+	// global vars, etc. as needed to run the tests below
+	Initialize(fakePGOClient)
+
+	// create a mock backrest repo secret with default values populated for the various S3
+	// annotations
+	mockBackRestRepoSecret := v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				config.ANNOTATION_S3_BUCKET:     "bucket",
+				config.ANNOTATION_S3_ENDPOINT:   "endpoint",
+				config.ANNOTATION_S3_REGION:     "region",
+				config.ANNOTATION_S3_URI_STYLE:  "path",
+				config.ANNOTATION_S3_VERIFY_TLS: "false",
+			},
+		},
+	}
+	defaultRestoreFromCluster := "restoreFromCluster"
+
+	type Env struct {
+		EnvVars []corev1.EnvVar
+	}
+
+	// test all env vars are properly set according the contents of an existing pgBackRest
+	// repo secret
+	t.Run("populate from secret", func(t *testing.T) {
+
+		backRestRepoSecret := mockBackRestRepoSecret.DeepCopy()
+		s3EnvVars := GetPgbackrestBootstrapS3EnvVars(defaultRestoreFromCluster, backRestRepoSecret)
+		// massage the results a bit so that we can parse as proper JSON to validate contents
+		s3EnvVarsJSON := strings.TrimSuffix(`{"EnvVars": [`+s3EnvVars, ",\n") + "]}"
+
+		s3Env := &Env{}
+		if err := json.Unmarshal([]byte(s3EnvVarsJSON), s3Env); err != nil {
```

Comment:
It looks like the local type might not be necessary.

```suggestion
		s3EnvVarsJSON := `[`+strings.TrimSuffix(s3EnvVars, ",\n") + `]`

		s3EnvVars := []corev1.EnvVar{}
		if err := json.Unmarshal([]byte(s3EnvVarsJSON), &s3EnvVars); err != nil {
```